### PR TITLE
Wrap internal embeds with div, add classes

### PR DESCRIPTION
**Pull request by trieloff**
In #408 @davidnuescheler asked for additional HTML decoration for internal embeds. This PR wraps internal embeds with a `div` with two classes:
- `embed-internal`
- `embed-internal-$basename` where base name is the base file name of the file included, so that including a `/components/copyright.md` would yield an `embed-internal-copyright` class name

This is breaking the diff tests (of course) and one integration test which I will fix.

### CODE REVIEW

**Comment by trieloff** [comment]
Labeled as breaking change. None of the top ten pages are affected, but @davidnuescheler should verify before we release this.

**Comment by davidnuescheler** [comment]
i am a bit suprised that we wouldn't have an `embed` class as well?

**Comment by kptdobe** [comment]
This might be a spec change but last night I thought about this: for theblog, we use internal embeds for promotions (like `/en/promotions/xyz.html`). Having a class `embed-internal-xyz` is totally useless in that case but `embed-internal-promotions` would allow to remove the `class: promotion` frontmatter requirement in the promotion files.

**Comment by trieloff** [comment]
@kptdobe: https://github.com/adobe/helix-pages/pull/417/commits/f888ae4cc5cfca8d57a63f10058ef6bfa7824a5a 999 more spec changes and we call it even.

File: src/embed_html.htl [hunk 1]
```diff
@@ -1 +1,3 @@
+<div class="embed embed-internal embed-internal-${content.meta.basename} embed-internal-${content.meta.dirame}-${content.meta.basename}">
```
Comment (kptdobe) [inline src/embed_html.htl:1]:
```suggestion
<div class="embed embed-internal embed-internal-${content.meta.basename} embed-internal-${content.meta.dirame}">
```

Nothing I can do with `embed-internal-${content.meta.dirame}-${content.meta.basename}` classname

File: test/unit/embed.html.pre.test.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          host: 'foo.bar',
+          'hlx-forwarded-host': 'www.foo.bar, foo-baz.hlx.page',
+        },
+        url: '/baz.html',
+        pathInfo: '/components/foo.bar.html',
+      },
+    };
+
+    pre(context);
+    assert.equal(context.content.meta.basename, 'foo');
```
Comment (kptdobe) [inline test/unit/embed.html.pre.test.js:47]:
```suggestion
    assert.equal(context.content.meta.basename, 'foo');
    assert.equal(context.content.meta.dirname, 'components');
```

For completeness.

File: test/unit/embed.html.pre.test.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          host: 'foo.bar',
+          'hlx-forwarded-host': 'www.foo.bar, foo-baz.hlx.page',
+        },
+        url: '/baz.html',
+        pathInfo: '/components/FOO<>.bar.html',
+      },
+    };
+
+    pre(context);
+    assert.equal(context.content.meta.basename, 'foo');
```
Comment (kptdobe) [inline test/unit/embed.html.pre.test.js:68]:
```suggestion
    assert.equal(context.content.meta.basename, 'foo');
    assert.equal(context.content.meta.dirname, 'components');
```

For completeness.

File: src/embed_html.htl (hunk 1 above)
Comment (kptdobe) [reply src/embed_html.htl:1]: Idea is to be able to address and style all promotions embeds, which is not doable if the class contains each filename.

**Pull request merged**